diff --git a/fhem/FHEM/98_SVG.pm b/fhem/FHEM/98_SVG.pm
index 434e79613..6f853922f 100644
--- a/fhem/FHEM/98_SVG.pm
+++ b/fhem/FHEM/98_SVG.pm
@@ -2692,7 +2692,7 @@ plotAsPng(@)
     <ul>
       <li>Normalerweise m&uuml;ssen SVG-Ger&auml;te nicht manuell erzeugt
         werden, da FHEMWEB es f&uuml;r den Nutzer einfach macht: man muss in
-        der Detailansicht eines FileLogs wechseln und auf "Create SVG instance"
+        die Detailansicht eines FileLogs wechseln und auf "Create SVG instance"
         klicken.</li>
 
       <li>CURRENT als &lt;logfile&gt; wird immer das aktuelle Logfile
@@ -2718,8 +2718,8 @@ plotAsPng(@)
       den Namen der SVG Instanz tr&auml;gt; bereits bestehende Dateien mit
       gleichem Namen werden &uuml;berschrieben. Diese Vorgehensweise ist
       notwendig, wenn man den Ploteditor benutzt. Erzeugt man aus der
-      Detailansicht des FileLogs die SVG Instanz, wird eine eindeutige
-      .gplot-Datei erzeugt. In diesem Fall ist dieses Befehl nicht
+      Detailansicht eines FileLogs die SVG Instanz, wird eine eindeutige
+      .gplot-Datei erzeugt. In diesem Fall ist dieser Befehl nicht
       erforderlich.</li>
 
   </ul><br>
