Index: FHEM/RESIDENTStk.pm
===================================================================
--- FHEM/RESIDENTStk.pm	(Revision 23459)
+++ FHEM/RESIDENTStk.pm	(Arbeitskopie)
@@ -127,7 +127,7 @@
 
     # Initialize the module and the device
     return $@ unless ( FHEM::Meta::SetInternals($hash) );
-    use version 0.77; our $VERSION = FHEM::Meta::Get( $hash, 'version' );
+    use version 0.77; my $VERSION = FHEM::Meta::Get( $hash, 'version' );
 
     $hash->{MOD_INIT}  = 1;
     $hash->{NOTIFYDEV} = "global";
