diff --git a/fhem/FHEM/44_TEK603.pm b/fhem/FHEM/44_TEK603.pm
index 074c8b3ea..58e174d2b 100644
--- a/fhem/FHEM/44_TEK603.pm
+++ b/fhem/FHEM/44_TEK603.pm
@@ -139,16 +139,28 @@ sub TEK603_undef($$) {
 }
 
 
-sub TEK603_ready($) {
+sub TEK603_ready() {
 	my ($hash) = @_;
 	my $name = $hash->{NAME};
 	return if (IsDisabled($name));
 	return DevIo_OpenDev($hash, 1, 'TEK603_doInit') if($hash->{STATE} eq 'disconnected');
 
 	# This is relevant for windows/USB only
+	my ($BlockingFlags, $InBytes, $OutBytes, $ErrorFlags);
 	my $po = $hash->{USBDev};
-	my ($BlockingFlags, $InBytes, $OutBytes, $ErrorFlags) = $po->status;
-	return ($InBytes > 0);
+
+	if ($po) {
+		($BlockingFlags, $InBytes, $OutBytes, $ErrorFlags) = $po->status;
+		return ($InBytes > 0);
+	}
+
+	# Someone set us up the bomb
+	Log3($hash->{NAME}, 1, qq[Can't read from $hash->{DeviceName}]);
+
+	# disable device
+	Log3($hash->{NAME}, 1, qq[Disabled device due read errors]);
+	CommandAttr(undef, $hash->{NAME} . ' disable 1');
+	return;
 }
 
 sub TEK603_read($) {
