diff --git a/fhem/FHEM/98_update.pm b/fhem/FHEM/98_update.pm
index ebb8584e5..03f5a5a2d 100644
--- a/fhem/FHEM/98_update.pm
+++ b/fhem/FHEM/98_update.pm
@@ -724,10 +724,10 @@ upd_writeFile($$$$)
   <ul>
     <a name="updateInBackground"></a>
     <li>updateInBackground<br>
-        Wenn dieses Attribut gesetzt ist, wird das update Befehl in einem
-        separaten Prozess ausgef&uuml;hrt, und alle Meldungen werden per Event
+        Wenn dieses Attribut gesetzt ist, wird der update Befehl in einem
+        separaten Prozess ausgef&uuml;hrt und alle Meldungen werden per Event
         &uuml;bermittelt. In der telnet Sitzung wird inform, in FHEMWEB wird
-        das Event Monitor aktiviert. Die Voreinstellung ist an, zum
+        der Event Monitor aktiviert. Die Voreinstellung ist an, zum
         Deaktivieren bitte Attribut auf 0 setzen.
         </li><br>
 
@@ -735,7 +735,7 @@ upd_writeFile($$$$)
     <li>updateNoFileCheck<br>
         Wenn dieses Attribut gesetzt ist, wird die Gr&ouml;&szlig;e der bereits
         vorhandenen, lokalen Datei nicht mit der Sollgr&ouml;&szlig;e
-        verglichen. Dieses Attribut wurde nach nicht genau spezifizierten Wnsch
+        verglichen. Dieses Attribut wurde nach nicht genau spezifiziertem Wunsch
         erfahrener FHEM Benutzer eingefuehrt, die Voreinstellung ist 0.
         </li><br>
 
@@ -754,7 +754,7 @@ upd_writeFile($$$$)
     <a name="exclude_from_update"></a>
     <li>exclude_from_update<br>
         Enth&auml;lt eine Liste durch Leerzeichen getrennter Dateinamen
-        (regexp), welche nicht im update ber&uuml;cksichtigt werden.<br>
+        (Regexp), welche nicht beim update ber&uuml;cksichtigt werden.<br>
         Falls der Wert commandref enth&auml;lt, dann wird commandref_join.pl
         nach dem update nicht aufgerufen, d.h. die Gesamtdokumentation ist
         nicht mehr aktuell. Die Moduldokumentation bleibt weiterhin aktuell.
