diff --git a/fhem/FHEM/HttpUtils.pm b/fhem/FHEM/HttpUtils.pm
index 057e4d23f..276f37690 100644
--- a/fhem/FHEM/HttpUtils.pm
+++ b/fhem/FHEM/HttpUtils.pm
@@ -5,9 +5,11 @@ package main;
 use strict;
 use warnings;
 use MIME::Base64;
+use File::MimeInfo;
 use Digest::MD5 qw(md5_hex);
 use vars qw($SSL_ERROR);
 
+
 # Note: video does not work for every browser (Forum #73214)
 my %ext2MIMEType= qw{
   bmp   image/bmp
@@ -22,7 +24,7 @@ my %ext2MIMEType= qw{
   png   image/png
   svg   image/svg+xml
   txt   text/plain
-
+  mp3   audio/mpeg
 };
 
 my $HU_use_zlib;
@@ -38,8 +40,16 @@ ext2MIMEType($) {
 sub
 filename2MIMEType($) {
   my ($filename)= @_;
-  $filename =~ m/^.*\.([^\.]*)$/;
-  return ext2MIMEType($1);
+
+  my $mimeType  = mimetype($filename);
+
+  unless ($mimeType);
+  {
+    my $extension =~ m/^.*\.([^\.]*)$/;
+    return ext2MIMEType($extension);
+  }
+
+  return $mimeType;
 }
   
 
