Index: FHEM/01_FHEMWEB.pm
===================================================================
--- FHEM/01_FHEMWEB.pm	(Revision 23511)
+++ FHEM/01_FHEMWEB.pm	(Arbeitskopie)
@@ -381,6 +381,8 @@
     }
   }
 
+  ::readingSingleUpdate($hash, q{bytes_read}, $sh->{BYTES_READ}, 1);
+
   if($hash->{websocket}) { # 59713
     # https://tools.ietf.org/html/rfc6455
     my $fin  = (ord(substr($hash->{BUF},0,1)) & 0x80)?1:0;
@@ -727,6 +729,7 @@
     }
   }
   $defs{$hash->{SNAME}}{BYTES_WRITTEN} += length($txt);
+  ::readingSingleUpdate($hash, q{bytes_written}, $defs{$hash->{SNAME}}{BYTES_WRITTEN}, 1);
   return addToWritebuffer($hash, $txt, $callback, $nolimit);
 }
 
