diff --git a/fhem/FHEM/10_MQTT_GENERIC_BRIDGE.pm b/fhem/FHEM/10_MQTT_GENERIC_BRIDGE.pm
index ee03ff081..6f1e6c726 100644
--- a/fhem/FHEM/10_MQTT_GENERIC_BRIDGE.pm
+++ b/fhem/FHEM/10_MQTT_GENERIC_BRIDGE.pm
@@ -2997,7 +2997,7 @@ sub onmessage($$$) {
    <li>
     <p>globalTypeExclude<br/>
         Defines (device) types and readings that should not be considered in the transmission.
-        Values can be specified separately for each direction (buplish or subscribe). Use prefixes 'pub:' and 'sub:' for this purpose.
+        Values can be specified separately for each direction (publish or subscribe). Use prefixes 'pub:' and 'sub:' for this purpose.
         A single value means that a device is completely ignored (for all its readings and both directions). 
         Colon separated pairs are interpreted as '[sub:|pub:]Type:Reading'. 
         This means that the given reading is not transmitted on all devices of the given type. 
@@ -3009,7 +3009,7 @@ sub onmessage($$$) {
    <li>
     <p>globalDeviceExclude<br/>
         Defines device names and readings that should not be transferred. 
-        Values can be specified separately for each direction (buplish or subscribe). Use prefixes 'pub:' and 'sub:' for this purpose.
+        Values can be specified separately for each direction (publish or subscribe). Use prefixes 'pub:' and 'sub:' for this purpose.
         A single value means that a device with that name is completely ignored (for all its readings and both directions).
         Colon-separated pairs are interpreted as '[sub:|pub:]Device:Reading'. 
         This means that the given reading is not transmitted to the given device.</p>
@@ -3392,7 +3392,7 @@ sub onmessage($$$) {
    <li>
     <p>globalTypeExclude<br/>
         Definiert (Geraete-)Typen und Readings, die nicht bei der Uebertragung beruecksichtigt werden. 
-        Werte koennen getrennt fuer jede Richtung (buplish oder subscribe) vorangestellte Prefixe 'pub:' und 'sub:' angegeben werden.
+        Werte koennen getrennt fuer jede Richtung (publish oder subscribe) vorangestellte Prefixe 'pub:' und 'sub:' angegeben werden.
         Ein einzelner Wert bedeutet, dass ein Geraet diesen Types komplett ignoriert wird (also fuer alle seine Readings und beide Richtungen).
         Durch ein Doppelpunkt getrennte Paare werden als [sub:|pub:]Type:Reading interptretiert. 
         Das Bedeutet, dass an dem gegebenen Type die genannte Reading nicht uebertragen wird.
@@ -3405,7 +3405,7 @@ sub onmessage($$$) {
    <li>
     <p>globalDeviceExclude<br/>
         Definiert Geraetenamen und Readings, die nicht uebertragen werden.
-        Werte koennen getrennt fuer jede Richtung (buplish oder subscribe) vorangestellte Prefixe 'pub:' und 'sub:' angegeben werden.
+        Werte koennen getrennt fuer jede Richtung (publish oder subscribe) vorangestellte Prefixe 'pub:' und 'sub:' angegeben werden.
         Ein einzelner Wert bedeutet, dass ein Geraet mit diesem Namen komplett ignoriert wird (also fuer alle seine Readings und beide Richtungen).
         Durch ein Doppelpunkt getrennte Paare werden als [sub:|pub:]Device:Reading interptretiert. 
         Das Bedeutet, dass an dem gegebenen Geraet die genannte Reading nicht uebertragen wird.</p>
